internal/v4/api.go

```diff
@@ -58,13 +59,14 @@ func New(store *charmstore.Store, config charmstore.ServerParams) *Handler {
 			"macaroon":           router.HandleJSON(h.serveMacaroon),
 		},
 		Id: map[string]router.IdHandler{
-			"archive":     h.serveArchive,
-			"archive/":    h.serveArchiveFile,
-			"diagram.svg": h.serveDiagram,
-			"expand-id":   h.serveExpandId,
-			"icon.svg":    h.serveIcon,
-			"readme":      h.serveReadMe,
-			"resources":   h.serveResources,
+			"archive":     h.idHandler(h.serveArchive),
```

Comment:
Perhaps it might be better to name "idHandler" after the
reason we're using it, so:

s/idHandler/authIdHandler/

?